1. Basic Web Development/script.ts

```diff
@@ -0,0 +1,405 @@
+        newButtonEdit.classList.add('edit-btn');
+
+        newElement.appendChild(newButtonEdit);
+        newElement.appendChild(newButtonDelete);
+        newElement.appendChild(newLabel);
+        newElement.appendChild(newInput);
+
+        permissionsList.appendChild(newElement);
+
+    });
+
+    this.BOXES = getBoxes();
+}
+
+function handleRadios() {
+    document.addEventListener('change', event => {
+        if ((event.target as HTMLInputElement).name == "preset") {
+            switch ((event.target as HTMLInputElement).value) {
+                case "All":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, true);
+                    this.changeBoxesState(this.BOXES.allBoxes, true);
+                    break;
+
+                case "Permissive":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, true);
+                    this.changeBoxesState(this.BOXES.allBoxes, false);
+                    break;
+
+                case "Necessary":
+                    this.changeBoxesState(this.BOXES.pmsBoxes, false);
+                    this.changeBoxesState(this.BOXES.allBoxes, false);
+                    break;
+
+                default:
+                    console.log("#ERROR");
+                    break;
+            }
+        }
+    });
+}
+
+function formSubmission(form: HTMLElement) {
+    document.addEventListener('submit', (event: Event) => {
+        event.preventDefault();
+
+        if ((event.target as HTMLElement).id == "permissions-form") {
+            submitPermissions(form);
+
+        } else if ((event.target as HTMLElement).id == "creation-form") {
+            submitCreation(form);
+
+        } else if ((event.target as HTMLElement).id == "edit-form") {
+            submitEdit(form);
+        }
+    });
+}
+
+function submitPermissions(form: HTMLElement) {
+    let messageData: { id: string, status: boolean }[] = [];
+
+    this.PERMISSIONS.forEach((element: Permission) => {
+        let checkedStatus = (document.getElementById(element.id) as HTMLInputElement).checked;
+
+        let newData = { "id": element.id, "status": checkedStatus };
+
+        messageData.push(newData);
+    });
+
+    const messageJSON = JSON.stringify(messageData);
+
+    fetch(API_URL, { method: 'POST', headers: HEADERS, body: messageJSON })
+        .then(response => response.json())
+        .then(data => {
+            console.log(data);
+        })
+        .catch((error) => {
+            console.error('Error:', error);
+            permissionsRoutine();
+        });
+}
+
+function submitCreation(form: HTMLElement) {
+    const type = document.querySelector('input[type=radio][name=presetC]:checked').id;
+    const text = (document.getElementById('permName') as HTMLInputElement).value;
+
+    if (!text) {
+        document.getElementById('text-alert').classList.remove('no-display');
+        return;
+    } else {
+        document.getElementById('text-alert').classList.add('no-display');
+    }
```

Comment:
```suggestion
    if (!text) {
        document.getElementById('text-alert').classList.remove('no-display');
        return;
    }
    
    document.getElementById('text-alert').classList.add('no-display'); 
```

since you're using return, you can drop the else